fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/db2/Db2ListTablesForSchema.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * (C) Copyright IBM Corp. 2021
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.ibm.fhir.database.utils.db2;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+
+import com.ibm.fhir.database.utils.api.IDatabaseSupplier;
+import com.ibm.fhir.database.utils.api.IDatabaseTranslator;
+import com.ibm.fhir.database.utils.common.DataDefinitionUtil;
+import com.ibm.fhir.database.utils.common.SchemaInfoObject;
+import com.ibm.fhir.database.utils.common.SchemaInfoObject.Type;
+
+/**
+ * DAO to fetch the names of tables in the given schema
+ */
+public class Db2ListTablesForSchema implements IDatabaseSupplier<List<SchemaInfoObject>> {
+    
+    // The schema of the table
+    private final String schemaName;
+
+    /**
+     * Public constructor
+     * @param schemaName
+     */
+    public Db2ListTablesForSchema(String schemaName) {
+        this.schemaName = DataDefinitionUtil.assertValidName(schemaName);
+    }
+
+    @Override
+    public List<SchemaInfoObject> run(IDatabaseTranslator translator, Connection c) {
```

Comment:
Seems very useful for the drop table migration step. https://github.com/IBM/FHIR/issues/713